first screen without segue

Cona7 opened the pull request.

Cona7 committed:
first screen without segue

Cona7 committed:
added navigation to second ccreen

Rep2 commented on WorkoutAppSnapKit/AppDelegate.swift:
```diff
@@ -8,10 +8,20 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+        let viewController = ProfileViewController(nibName: nil, bundle: nil)
+        let navigationController = UINavigationController(rootViewController: viewController)
+
+        navigationController.navigationBar.barStyle = .black
```
Find a better solution

Cona7 replied:
Found better solution through info.plist

Rep2 commented on WorkoutAppSnapKit/ProfileViewController.swift:
```diff
@@ -0,0 +1,151 @@
+
+    struct CellModel {
+        let name: String
+        let number: String = "8545"
```
`: String` not needed

Rep2 commented on WorkoutAppSnapKit/ProfileViewController.swift:
```diff
@@ -0,0 +1,151 @@
+    lazy var profileImageView: UIImageView = {
+        let image = #imageLiteral(resourceName: "oval2")
+        let imageView = UIImageView(image: image)
+        imageView.translatesAutoresizingMaskIntoConstraints = false
```
I think the isn't needed when using SnapKit

Rep2 commented on WorkoutAppSnapKit/ProfileViewController.swift:
```diff
@@ -0,0 +1,151 @@
+
+    let helveticaNeueFontString = "HelveticaNeue"
+
+    let tableView = UITableView(frame: CGRect.zero, style: .grouped)
```
Change to lazy

Rep2 commented on WorkoutAppSnapKit/ProfileViewController.swift:
```diff
@@ -0,0 +1,151 @@
+        let attributes = [NSAttributedString.Key.font: UIFont(name: helveticaNeueFontString, size: 17)!]
+        UINavigationBar.appearance().titleTextAttributes = attributes
+
+        tableView.delegate = self
```
Move to tableView lazy computed property

Rep2 commented on WorkoutAppSnapKit/ProfileViewController.swift:
```diff
@@ -0,0 +1,151 @@
+        navigationItem.rightBarButtonItem = settingsBarItem
+
+        let attributes = [NSAttributedString.Key.font: UIFont(name: helveticaNeueFontString, size: 17)!]
+        UINavigationBar.appearance().titleTextAttributes = attributes
```
Move to AppDelegate

Rep2 replied:
This also might not be needed

Rep2 commented on WorkoutAppSnapKit/ProfileViewController.swift:
```diff
@@ -0,0 +1,151 @@
+
+        cell.stringLabel.text = workoutCell.name
+        cell.numberLabel.text = workoutCell.number
+        cell.stringLabel.letterSpace = 0.4
```
Can this be moved to cell?

Cona7 replied:
I can achieve a letter spacing in this step only, when I'm initializing cells, because of extension I'm using.

Rep2 commented on WorkoutAppSnapKit/TypeViewCell.swift:
```diff
@@ -0,0 +1,56 @@
+        let label = UILabel()
+        label.textColor = .customBlue
+        label.font = UIFont(name: "Helvetica-Neue", size: 17.0)
+        label.textAlignment = NSTextAlignment.right
```
`label.textAlignment = .right`

Cona7 committed:
PR code refactor

- refactor NSAttributerdString in AppDelegate class
- set light content status bar through info.plist
- extended UIFont
- PR corrections

Rep2 commented on WorkoutAppSnapKit/EditProfileViewController.swift:
```diff
@@ -0,0 +1,32 @@
+        return imageView
+    }()
+
+    lazy var leftProfileImageView = profileImageView
```
Both imageViews hold reference to the same thing. You could do it like this.

```
var imageView: UIImageView {
        let image = #imageLiteral(resourceName: "oval2")
        let imageView = UIImageView(image: image)
        imageView.translatesAutoresizingMaskIntoConstraints = false
        imageView.contentMode = .scaleAspectFit
        return imageView
}

lazy var profileImageView = imageView
lazy var leftProfileImageView = imageView
```

Cona7 replied:
Yes, I did it as you said...I made computed property and assigned it to lazy variables. But, I made these corrections on another branch, but I forgot to update it here, my bad. I had also omitted line:

`imageView.translatesAutoresizingMaskIntoConstraints = false`

 from review part below.

I will try to inspect all classes before the push

Rep2 commented on WorkoutAppSnapKit/EditProfileViewController.swift:
```diff
@@ -0,0 +1,32 @@
+    lazy var profileImageView: UIImageView = {
+        let image = #imageLiteral(resourceName: "oval2")
+        let imageView = UIImageView(image: image)
+        imageView.translatesAutoresizingMaskIntoConstraints = false
```
Did you check this?

Cona7 committed:
refactor second screen code
